samples/CosmosDbIntegtation/ReadThroughSamples/StreamSample.cs

```diff
@@ -0,0 +1,79 @@
+        /// </summary>
+        /// <param name="entry"> The message which has gone through the stream. Includes message id alongside the key/value pairs </param>
+        /// <param name="items"> Container for where the CosmosDB items are stored </param>
+        /// <param name="logger"> ILogger used to write key information </param>
+        /// <returns></returns>
+        [FunctionName(nameof(ReadThroughForStreamSingleDocumentAsync))]
+        public static async Task ReadThroughForStreamSingleDocumentAsync(
+               [RedisPubSubTrigger(redisConnectionSetting, "__keyevent@0__:keymiss")] string entry,
+               [CosmosDB(
+                    databaseName: databaseSetting,
+                    containerName: containerSettingSingleDocument,
+                    Connection = cosmosDbConnectionSetting)] CosmosClient cosmosDbClient,
+               ILogger logger)
+        {
+            if (entry == streamNameSingleDocument)
+            {
+                // Connect CosmosDB container
+                Cosmos.Container cosmosDbContainer = cosmosDbClient.GetDatabase(Environment.GetEnvironmentVariable(databaseSetting.Substring(1, databaseSetting.Length - 2)))
+                    .GetContainer(Environment.GetEnvironmentVariable(containerSettingSingleDocument.Substring(1, containerSettingSingleDocument.Length - 2)));
+
+                // Query Database by the stream name
+                FeedIterator<StreamDataSingleDocument> query = cosmosDbContainer.GetItemLinqQueryable<StreamDataSingleDocument>(true)
+                                     .Where(b => b.id == streamNameSingleDocument)
+                                     .ToFeedIterator();
+
+                FeedResponse<StreamDataSingleDocument> response = await query.ReadNextAsync();
+                StreamDataSingleDocument results = response.FirstOrDefault(defaultValue: null);
+
+                //If stream note found
+                if (results == null)
+                {
+                    logger.LogInformation("stream not found");
+                }
+                else
+                {
+                    // Go through each message and format the key/value pairs
+                    foreach (var message in results.messages)
+                    {
+                        var values = new NameValueEntry[message.Value.Count];
+                        int i = 0;
+                        foreach (var pair in message.Value)
+                        {
+                            values[i++] = new NameValueEntry(pair.Key, pair.Value);
+                        }
+
+                        // Upload value to Redis Stream
+                        await redisDB.Value.StreamAddAsync(streamNameSingleDocument, values, messageId: message.Key, maxLength: results.maxlen);
+
+                    }
+                }
+            }

```

Comment:
Should there be an error/warn log in the `else` case? Seems like people shouldn't be calling this with the wrong `entry`?

Reply:
Or an exception as well maybe?